### PR TITLE
feat: enrich review comments with clickable file:line links

### PR DESCRIPTION
**Pull request by CreatibOfficiel**
## Summary

- Replace plain `file.py:42` references in `POST_COMMENT` bodies with clickable GitLab blob links pointing to the exact line at the head SHA
- Add `baseUrl` to `ExecutionContext` and thread it through the controller → executor → gateway pipeline
- Extract base URL from repository remote URL (supports both HTTPS and SSH formats)
- Strip `CLAUDECODE` env var when spawning Claude child process to allow nested invocations
- Normalize SSH git URLs to HTTPS in config loader

## Test plan

- [x] 12 unit tests for `commentLinkEnricher` covering: simple refs, backtick-wrapped, parenthesized, nested paths, URL exclusion, multiple refs
- [x] TypeScript build passes (0 errors)
- [x] Full test suite: 925 pass, 0 new regressions (2 pre-existing CLI integration failures unrelated)
- [ ] Manual: trigger a review on a GitLab MR and verify the summary comment contains clickable file links

### CODE REVIEW

File: src/interface-adapters/controllers/webhook/gitlab.controller.ts [hunk 1]
```diff
@@ -30,6 +30,24 @@ import type { ReviewContextGateway } from '@/entities/reviewContext/reviewContex
 import type { ThreadFetchGateway } from '@/entities/threadFetch/threadFetch.gateway.js';
 import type { DiffMetadataFetchGateway } from '@/entities/diffMetadata/diffMetadata.gateway.js';
 
+function extractBaseUrl(remoteUrl: string): string | undefined {
```
Comment (DGouron) [inline src/interface-adapters/controllers/webhook/gitlab.controller.ts]: 🚨 **[BLOCKING] Missing tests for `extractBaseUrl` pure function**

📍 `gitlab.controller.ts:33-50`

`extractBaseUrl` is a pure function with 3 code paths (HTTPS URL, SSH URL, invalid/fallback). It has zero tests.

> "Never write production code without a failing test first." — CLAUDE.md

**Fix**: Create `src/tests/units/interface-adapters/controllers/webhook/extractBaseUrl.test.ts` covering:
1. HTTPS URL → extracts protocol + host
2. SSH URL (`git@host:org/repo.git`) → extracts `https://host`
3. Invalid URL → returns `undefined` (or better: `null`)
4. Edge cases: trailing slashes, `.git` suffix

**Recommendation**: Run `/tdd` to create tests with RED-GREEN-REFACTOR cycle.

File: src/services/commentLinkEnricher.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    if (/:\/{1,2}$/.test(beforeMatch) || /:\/{1,2}[\w.-]*$/.test(beforeMatch)) {
+      return match
+    }
+
+    // Skip if the filePath portion doesn't look like a real file (must contain a dot for extension)
+    // Already handled by regex, but double check for edge cases
+    if (!filePath.includes('.')) {
+      return match
+    }
+
```
Comment (DGouron) [inline src/services/commentLinkEnricher.ts]: ⚠️ **[IMPORTANT] GitLab-specific blob URL pattern — not compatible with GitHub**

📍 `commentLinkEnricher.ts:41`

The URL pattern `${baseUrl}/${projectPath}/-/blob/${headSha}/${filePath}#L${line}` uses GitLab's `/-/blob/` convention. GitHub uses `/blob/` (without the `/-/` prefix).

Since the project supports both GitLab and GitHub (see `GitHubReviewActionCliGateway`), this enricher will produce broken links for GitHub PRs.

**Fix**: Either:
1. Pass a `platform` parameter and build the URL accordingly:
   - GitLab: `/-/blob/`
   - GitHub: `/blob/`
2. Or make the URL template configurable via a parameter.

This isn't blocking the merge since the enrichment currently only triggers in the GitLab gateway, but it should be addressed before extending to GitHub.

**Comment by DGouron** [comment]
Hello, thank you for your contribution :)

**Comment by CreatibOfficiel** [comment]
## Review fixes applied

All review items addressed in commit cff187a:

| # | Status | Fix |
|---|--------|-----|
| 1 | Done | Relative imports → `@/` alias (2 files) |
| 2 | Done | `extractBaseUrl` exported + 7 unit tests |
| 3 | Done | `normalizeGitUrl` exported + 5 unit tests |
| 4 | Done | `baseUrl` type: `undefined` → `null` across gateway, executor, controller |
| 5 | Done | `platform` param on `enrichCommentWithLinks` + 3 tests for GitHub URL pattern |
| 6 | Done | `CLAUDE.md` doc updated (type + example URL) |

**Verification:** 942 tests pass (120 files), 0 regressions, typecheck clean.

**Comment by DGouron** [comment]
# Follow-up Review — PR #106

**Date**: 2026-03-06
**Previous review**: Initial review on PR #106

---

## Blocking Issues Verification

| # | Issue | Status | Comment |
|---|-------|--------|---------|
| 1 | Relative import in `reviewAction.gitlab.cli.gateway.ts:4` | ✅ FIXED | Now uses `@/services/commentLinkEnricher.js` |
| 2 | Relative import in `commentLinkEnricher.test.ts:2` | ✅ FIXED | Now uses `@/services/commentLinkEnricher.js` |
| 3 | Missing tests for `extractBaseUrl` | ✅ FIXED | 7 test cases covering HTTPS, HTTP, SSH, port, nested groups, empty, invalid |
| 4 | Missing tests for `normalizeGitUrl` | ✅ FIXED | 5 test cases covering SSH→HTTPS, .git strip, nested groups, pass-through |

## Important Issues Verification

| # | Issue | Status | Comment |
|---|-------|--------|---------|
| 5 | `extractBaseUrl` returns `undefined` instead of `null` | ✅ FIXED | Signature now `string \| null`, returns `null` on failure |
| 6 | GitLab-specific blob URL pattern incompatible with GitHub | ✅ FIXED | Added `platform` parameter (`'gitlab' \| 'github'`), tests cover both |
| 7 | Type assertion `null as string \| null` in production code | ✅ FIXED | `threadActionsExecutor.ts:45` now uses plain `null` without assertion |

---

## New Problems Detected

No new problems detected in the modifications.

---

## Final Verdict

| Criterion | Status |
|-----------|--------|
| Blocking issues fixed | 4/4 |
| Important issues fixed | 3/3 |
| New blockers | 0 |
| **Ready to merge** | ✅ Yes |

[REVIEW_STATS:blocking=0:warnings=0:suggestions=0:score=9]
